test/test_data/firefox.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2022 Orange
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+mapping_test_data = [
+    {
+        "product": "firefox",
+        "version": "106.0.3-1",
+        "version_strings": ["firefox-106.0.3-1.fc38"],
```

Comment:
The string here should be changed to strings that can be matched by atleast one pattern in `VERSION_PATTERNS`

eg: `r"ENCODER=Mozilla106.0(?:(?:\r?\n.*?)*)CN=Firefox U2F Soft Token"` with the version being `106.0`.